packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -686,6 +686,42 @@ export interface AutoScalingGroupProps extends CommonAutoScalingGroupProps {
    * @default false
    */
   readonly requireImdsv2?: boolean;
+
+  /**
+   * Specifies the upper threshold as a percentage of the desired capacity of the Auto Scaling group.
+   * It represents the maximum percentage of the group that can be in service and healthy, or pending,
+   * to support your workload when replacing instances.
+   *
+   * Value range is 0 to 100. After it's set, both `instanceMaintenancePolicyMinHealthPercentage` and
+   * `instanceMaintenancePolicyMaxHealthPercentage` to -1 will clear the previously set value.
+   *
+   * Both or neither of `instanceMaintenancePolicyMinHealthPercentage` and `instanceMaintenancePolicyMaxHealthPercentage`
+   * must be specified, and the difference between them cannot be greater than 100. A large range increases
+   * the number of instances that can be replaced at the same time.
+   *
+   * @see https://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-instance-maintenance-policy.html
+   *
+   * @default - Do not provide percentage.
+   */
+  readonly instanceMaintenancePolicyMaxHealthPercentage?: number;
```

Comment:
```suggestion
   * Both or neither of `maintenancePolicyMaxHealthPercentage` and `maintenancePolicyMinHealthPercentage`
   * must be specified, and the difference between them cannot be greater than 100. A large range increases
   * the number of instances that can be replaced at the same time.
   *
   * @see https://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-instance-maintenance-policy.html
   *
   * @default - No instance maintenance policy.
   */
  readonly maintenancePolicyMaxHealthPercentage?: number;
```
Let's keep the variable name shorter since it's already pretty long.